### PR TITLE
chore: Update CI

### PR DESCRIPTION
**Pull request by metcoder95**
The PR aims to update the CI manifest and `dependabot` scope and cadency to up-to-date settings.
It introduces the following changes:

- chore: update CI
- chore: update dependabot manifest
- chore: sync lockfile

### CODE REVIEW

**Comment by metcoder95** [comment]
Reverted to `12`, `14`, and `16` as it seems there are changes required to support `18` onwards. I'll address it in a subsequent PR targeting a new branch for dropping support for old Node versions.

cc: @jasnell @RafaelGSS

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -8,30 +8,43 @@ jobs:
...
   test:
+    needs: [lint]
     name: Test
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        node-version: [12.x, 14.x, 16.x, 17.x]
-    runs-on: ${{matrix.os}}
+        node-version: [12.x, 14.x, 16.x]
```
Comment (RafaelGSS) [inline .github/workflows/nodejs.yml]: Why don't drop 12 and 14 here?

Comment (metcoder95) [reply .github/workflows/nodejs.yml]: I was considering it to add the whole drop of v12-14 within the work on `next` branch, as I think the drop should also sync with a drop of the supported version within `engines` of `package.json`. Wdyt?

**Comment by metcoder95** [comment]
@RafaelGSS, do you have an idea of from where the `13` is coming?
I tried to see where this is coming from but couldn't reach a conclusion. The `nodejs.yml` does not contains the v13 within the matrix

**Review by RafaelGSS — COMMENTED** [review]
@metcoder95 I think you can target next and drop v12, v14. So I can release a new major too

**Comment by metcoder95** [comment]
SGTM, it's done now 👍

**Pull request merged**